docs/csharp/tutorials/mixins-with-default-interface-methods.md

```diff
@@ -0,0 +1,121 @@
+
+Starting with C# 8.0, you can declare the default implementations as interface methods. Then, every class automatically uses the default implementation. Any class that can provide a better implementation can override the interface method definition with a better algorithm. In one sense, this technique sounds similar to how you could use [extension methods](../programming-guide/classes-and-structs/extension-methods.md).
+
+In this article, you'll learn how default interface implementations enable new scenarios.
+
+## Design the application
+
+Consider a home automation application. You probably have many different types of lights and indicators that could be used throughout the house. Every light must support APIs to turn them on and off, and to report the current state. Some lights and indicators may support other features, such as:
+
+- Turn light on, then turn it off after a timer.
+- Blink the light for a period of time.
+
+Some of these extended capabilities could be emulated in devices that support the minimal set. That indicates providing a default implementation. For those devices that have more capabilities built in, the device software would use the native capabilities. For other lights, they could choose to implement the interface and use the default implementation.
+
+Default interface members is a better solution for this scenario than extension methods. Class authors can control which interfaces they choose to implement. Those interfaces they choose are available as methods. In addition, because default interface methods are virtual by default, the method dispatch always chooses the implementation in the class. 
+
+Let's create the code to demonstrate these differences.
+
+## Create interfaces
+
+Start by creating the interface that defines the behavior for all lights:
+
+[!code-csharp[Declare base interface](~/samples/csharp/tutorials/mixins-with-interfaces/UnusedSampleCode.cs?name=SnippetILightInterfaceV1)]
+
+A basic overhead light fixture might implement this interface as shown in the following code:
+
+[!code-csharp[First overhead light](~/samples/csharp/tutorials/mixins-with-interfaces/UnusedExampleCode.cs?name=SnippetOverheadLightV1)]
+
+In this tutorial, the code doesn't drive IoT devices, but emulates those activities by writing messages to the console. You can explore the code without automating your house.
+
+Next, let's define the interfaces for a light that can automatically turn off after a timeout:
+
+[!code-csharp[pure Timer interface](~/samples/csharp/tutorials/mixins-with-interfaces/UnusedExampleCode.cs?name=SnippetPureTimerInterface)]
+
+You could add a basic implementation to the overhead light, but a better solution is to modify this interface definition to provide a `virtual` default implementation:
+
+[!code-csharp[Timer interface](~/samples/csharp/tutorials/mixins-with-interfaces/ITimerLight.cs?name=SnippetTimerLightFinal)]
```

Comment:
minor nit: this code has a sentence that says:
"Using default interface implementation for ITimerLight.TurnOnFor"

wondering if it would be better as "Using the default interface implementation for ITimerLight.TurnOnFor" or
"Using the default interface implementation for the ITimerLight.TurnOnFor method"